src/vehicle_display.cpp

```diff
@@ -120,6 +120,114 @@ nc_color vehicle::part_color( const int p, const bool exact ) const
     }
 }
 
+size_t vehicle::vehicle_info( std::vector<std::string> &out_info, const uint32_t veh_part,
+                              const uint32_t max_width, const int hl /* = -1*/ )
+{
+    const size_t old_size = out_info.size();
+
+    //brief description
+    std::string message = string_format( _( "There is a %s there. Parts:" ), name );
+    foldstring( out_info, colorize( message, c_dark_gray ), max_width );
+    const size_t desc_offset = out_info.size() - old_size;
+
+    //boundary check for part id
+    if( veh_part >= parts.size() ) {
+        return 0;
+    }
+
+    std::vector<int> parts_id_list = parts_at_relative( parts[veh_part].mount, true );
+
+    for( size_t i = 0; i < parts_id_list.size(); i++ ) {
+
+        const size_t id = static_cast<size_t>( parts_id_list[i] );
+        const vehicle_part &vp = parts[id];
+        //color
+        nc_color sym_clr = c_light_gray;
+        nc_color part_clr = vp.is_broken() ? c_dark_gray : vp.get_base().damage_color();
+        //name
+        std::string partname = vp.name() + " ";
+        //fuel content
+        partname += ( vp.is_fuel_store() && vp.ammo_current() != "null" ) ? fuel_info( vp ) : "";
+        //cargo
+        partname += part_flag( id, "CARGO" ) ? cargo_info( id ) : "";
+        //armor
+        auto sym = armor_info( id );
+        //highlight selected
+        if( static_cast<const int>( i ) == hl ) {
+            sym_clr = hilite( c_light_gray );
+            part_clr = hilite( part_clr );
+        }
+
+        sym.first = colorize( sym.first, sym_clr );
+        sym.second = colorize( sym.second, sym_clr );
+        partname = colorize( partname, part_clr );
+        partname = trim_to_width( ( sym.first + partname + sym.second ), max_width );
+        out_info.push_back( partname );
+    }
+
+    //Covered or not?
+    size_t line_written = ( out_info.size() - old_size );
+    if( line_written > 0 ) {
+
+        const bool is_inside = vpart_position( *this, parts_id_list[0] ).is_inside();
+        std::string desc = is_inside ? _( "Interior" ) : _( "Exterior" );
+
+        const size_t target = out_info.size() - line_written;
+        std::string &line = out_info[target + desc_offset];
+
+        const size_t lenght = static_cast<size_t>( utf8_width( desc ) );
+        const size_t start_pos = lenght > max_width ? max_width : max_width - lenght;
+        const size_t naked_size = remove_color_tags( line ).size();
+
+        if( naked_size > start_pos ) {
```

Comment:
Wouldn't it have been easier to just add the Interior/Exterior tag on the first pass through parts_at_relative?
ie,
```
    for( size_t i = 0; i < parts_id_list.size(); i++ ) {
        //name
￼        std::string partname = vp.name() + " ";
        if( i == 0 ) {
            partname += parts[veh_part].inside ? _( "Interior" ) : _( "Exterior" );
        }
￼```

Reply:
you are doing a redundant check inside the for loop on every iteration, the if( i == 0 ).
It would have been easier, but I take that check outside the loop on purpose since it needs to happen only once

Reply:
I would rather have the code inside the loop and easy to understand and maintain than mildly faster and confusing.

Reply:
this is how that code looks now:

`    if( parts_id_list.size() ) {

        refresh_insides();
        std::string desc = parts[veh_part].inside ? _( "Interior" ) : _( "Exterior" );

        std::string &line = out_info[old_size + desc_offset];

        const size_t length = static_cast<size_t>( utf8_width( desc ) );
        const size_t start_pos = length > max_width ? max_width : max_width - length;
        const size_t naked_size = remove_color_tags( line ).size();

        if( naked_size > start_pos ) {
            line = trim_to_width( line, start_pos );
        } else {
            std::string padding( ( start_pos - naked_size ), ' ' );
            line += padding;
        }
        line += colorize( desc, c_light_gray );
    }`

The added complexity is not coming from the fact it is outside of the for loop, since that is adding only two lines of work: `std::string &line = out_info[old_size + desc_offset];`  required to get a reference of the correct line in which the "Interior"/"Exterior" text should go, and the "remove_color_tags()" required for knowing the true lenght of the text already in that line.
The added complexity is due to the fact this used to be a simple print on screen so you could just print it at ( x , y ) and it would overwrite wathever part name it was in there, and require no additional work. Being a string, it require the extra work/calculation of knowing how much space the "Interior/Exterior" text will need, adding the whitespace padding inbetween, account for some crazy translation that might be wider than the window in which this appear and trimming the part name at that line to the minimum required to display "Interior/Exterior" text at the end, and also colorizing it.
Even bringing that back into the for loop and getting rid of 2 lines wouldn't change much at this point